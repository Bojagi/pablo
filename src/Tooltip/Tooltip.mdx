---
name: Tooltip
route: /tooltip
menu: Components
---

import { Props, Playground } from 'docz'
import { Box } from '../Box/Box.tsx'
import { Tooltip } from './Tooltip.tsx'
import { Button } from '../Button/index.ts'
import { css } from '@emotion/react'

# Tooltip

A component that decorates it's child with a tooltip that shows hints/information text on hovering over the child.

## Examples

### Plain tooltip

<Playground>
  <Tooltip content="A tooltip shows on hovering (you just witnessed)">
    <Button color="plain">
      Hover over me
    </Button>
  </Tooltip>
</Playground>

### Tooltip positioning

A tooltip can be positioned on each of the for sides of it's child. 

<Playground>
  <Box display="flex" mx={-4}>
    <Tooltip side="left" content="Left tooltip">
      <Button mx={4} color="plain">
        Left
      </Button>
    </Tooltip>
    <Tooltip side="top" content="Top tooltip">
      <Button mx={4} color="plain">
        top
      </Button>
    </Tooltip>
    <Tooltip side="bottom" content="Bottom tooltip">
      <Button mx={4} color="plain">
        bottom
      </Button>
    </Tooltip>
    <Tooltip side="right" content="Right tooltip">
      <Button mx={4} color="plain">
        right
      </Button>
    </Tooltip>
  </Box>
</Playground>

### Tooltips with a delay

If you want to only show a tooltip after hovering over it for a certain time, you can use the `delay` prop and pass
a number that represents a millisecond time. This is useful for tooltips for more obvious actions.

<Playground>
  <Tooltip content="Sorry for being late" delay={500}>
      <Button color="plain">
        Quickly, hover over me!
      </Button>
    </Tooltip>
</Playground>

### Usage on disabled children

When a child has the `disabled` prop set to true, the tooltip doesn't show as well. If you want to get around this
behavior you need to wrap the child in a `div`, `span` or any other element. 

<Playground>
  <Tooltip content="I'm not there!">
    <Button mx={5} color="plain" disabled={true}>
      I am not clickable
    </Button>
  </Tooltip>
</Playground>

### Customizing the tooltip

Additionally to the component configuration, you can overwrite styles freely (like with every other Pablo component).

<Playground>
  <Tooltip
    content="I am very special"
    customStyles={{
      box: css`
        background-color: red;
        border-radius: 100px;
      `,
      arrow: css`
        border-bottom-color: red;
      `,
    }}
  >
    <Button color="plain">
      Hover over me ðŸŒˆ
    </Button>
  </Tooltip>
</Playground>
  

## Import

```js
import { Tooltip } from '@bojagi/pablo/Tooltip';
```

## Props

<Props of={Tooltip} />
