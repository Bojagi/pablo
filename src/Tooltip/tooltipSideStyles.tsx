import { css } from '@emotion/react';
import { getSpacing } from '../styleHelpers/getSpacing';

export const topArrowStyles = css`
  left: 50%;
  transform: scaleY(0.5) translateY(-50%) translateX(-50%) rotateZ(180deg);
`;

export const bottomArrowStyles = (props) => css`
  top: -${getSpacing(5)(props)};
  left: 50%;
  transform: scaleY(0.5) translateY(-50%) translateX(-50%) rotateZ(0deg);
`;

export const rightArrowStyles = (props) => css`
  left: -${getSpacing(5)(props)};
  top: 50%;
  transform: scale(0.5) translateX(-50%) translateY(-100%) rotateZ(270deg);
`;

export const leftArrowStyles = (props) => css`
  right: -${getSpacing(5)(props)};
  top: 50%;
  transform: scale(0.5) translateX(50%) translateY(-100%) rotateZ(90deg);
`;
