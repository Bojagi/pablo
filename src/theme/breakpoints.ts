export type Breakpoint = 'sm' | 'md' | 'lg' | 'xl';

export const breakpointNames: Breakpoint[] = ['sm', 'md', 'lg', 'xl'];

export type BreakpointsArray = [number, number, number, number];

export type Breakpoints = BreakpointsArray &
  Record<Breakpoint, number> & {
    breakpointNames: Breakpoint[];
  };

export function createBreakpoints(
  array: BreakpointsArray,
  usedBreakpointNames: Breakpoint[] = breakpointNames
): Breakpoints {
  const newBreakpoints = ([...array] as unknown) as Breakpoints;
  usedBreakpointNames.forEach((bp, index) => {
    newBreakpoints[bp] = newBreakpoints[index];
  });
  newBreakpoints.breakpointNames = usedBreakpointNames;

  return newBreakpoints as Breakpoints;
}

export const defaultBreakpoints: BreakpointsArray = [600, 1000, 1200, 1920];

export const breakpoints = createBreakpoints(defaultBreakpoints, breakpointNames);
