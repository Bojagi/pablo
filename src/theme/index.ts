export * from './defaultTheme';
export * from './context';
export * from './PabloThemeProvider';
export * from './useComponentStyle';
