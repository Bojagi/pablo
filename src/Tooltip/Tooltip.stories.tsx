import React from 'react';
import { css } from 'styled-components';
// eslint-disable-next-line import/no-extraneous-dependencies
import { Flex } from '../Box';
import { Button } from '../Button';
import { Tooltip } from './Tooltip';

export default {
  title: 'Tooltip',
};

const BaseStory = (args) => (
  <Flex height="100%" width="100vw" alignItems="center" justifyContent="center">
    <Tooltip content="Top tooltip" side="left" {...args}>
      <Button mx={5} color="plain">
        left
      </Button>
    </Tooltip>
    <Tooltip content="Top tooltip" side="top" {...args}>
      <Button mx={5} color="plain">
        {' '}
        top
      </Button>
    </Tooltip>
    <Tooltip content="Top tooltip" side="bottom" {...args}>
      <Button mx={5} color="plain">
        bottom
      </Button>
    </Tooltip>
    <Tooltip content="Top tooltip" side="right" {...args}>
      <Button mx={5} color="plain">
        right
      </Button>
    </Tooltip>
  </Flex>
);

export const WithoutDelay = BaseStory.bind(null);
WithoutDelay.args = {};

export const WithDelay = BaseStory.bind(null);
WithDelay.args = { delay: 400 };

export const WithDisabledButton = ({ disabled, ...args }) => (
  <Flex height="100%" width="100vw" alignItems="center" justifyContent="center">
    <Tooltip content="Top tooltip" side="top" {...args}>
      <Button mx={5} color="plain" disabled={disabled}>
        {' '}
        top
      </Button>
    </Tooltip>
  </Flex>
);
WithDisabledButton.args = {
  disabled: true,
};

export const WithCustomStyles = BaseStory.bind(null);
WithCustomStyles.args = {
  customStyles: {
    box: css`
      background-color: red;
      border-radius: 100px;
    `,
    arrow: css`
      border-bottom-color: red;
    `,
  },
};
