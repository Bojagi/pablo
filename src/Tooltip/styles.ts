import { css } from '@emotion/react';
import { getSpacing } from '../styleHelpers/getSpacing';
import { themeVars } from '../theme/themeVars';
import { Style } from '../theme/types';
import { BaseStyles } from '../types';

export type TooltipStyleProperties = 'box' | 'arrow';

export interface TooltipStyles extends BaseStyles<TooltipStyleProperties> {
  gap: Style;
  padding: Style;
  borderRadius: number;
  backgroundColor: Style;
  color: Style;
  zIndex: number;
}

export const tooltipStyles: TooltipStyles = {
  gap: getSpacing(2),
  padding: (props) => css`
    ${getSpacing(2)(props)} ${getSpacing(4)(props)}
  `,
  zIndex: 100,
  backgroundColor: themeVars.colors.common.black,
  color: themeVars.colors.common.white,
  borderRadius: 4,
};
