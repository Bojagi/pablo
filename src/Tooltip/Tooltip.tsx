import React, { ComponentElement, ComponentType, useEffect, useState } from 'react';
import styled from '@emotion/styled';
import { LayoutBoxProps } from '../Box';
import { Typography } from '../Typography';
import { conditionalStyles, getComponentStyle } from '../styleHelpers';
import { getSpacing } from '../styleHelpers/getSpacing';
import {
  topArrowStyles,
  bottomArrowStyles,
  rightArrowStyles,
  leftArrowStyles,
} from './tooltipSideStyles';
import { Popover } from '../Popover/Popover';
import { useComponentStyle } from '../theme/useComponentStyle';
import { BaseProps } from '../types';
import { TooltipStyleProperties } from './styles';
import { getCustomStyles } from '../utils/useCustomStyles';
import { baseStyle } from '../shared/baseStyle';
import { SlideAnimation } from '../animation/SlideAnimation';
import type { AnimationSetupProps, InOutAnimationProps } from '../animation';

export type TooltipSide = 'top' | 'left' | 'bottom' | 'right';

export interface TooltipProps<A extends object = object>
  extends LayoutBoxProps,
    BaseProps<TooltipStyleProperties> {
  content: React.ReactNode;
  side?: TooltipSide;
  showOnClick?: boolean;
  animation?: ComponentType<InOutAnimationProps<A>>;
  animationProps?: Partial<AnimationSetupProps & A>;
  delay?: number;
  disabled?: boolean;
  children: ComponentElement<any, any>;
}

interface TooltipPopoverProps extends BaseProps<TooltipStyleProperties> {
  side: TooltipSide;
}

const TooltipPopover = styled.div<TooltipPopoverProps>`
  ${baseStyle}
  z-index: ${getComponentStyle(['tooltip', 'zIndex'])};
  pointer-events: none;
  position: relative;
  border-radius: ${getComponentStyle(['tooltip', 'borderRadius'])}px;
  padding: ${getComponentStyle(['tooltip', 'padding'])};
  background-color: ${getComponentStyle(['tooltip', 'backgroundColor'])};
  color: ${getComponentStyle(['tooltip', 'color'])};
  white-space: nowrap;

  /* Bottom Arrow */
  &:after {
    ${baseStyle}
    content: '';
    position: absolute;

    ${conditionalStyles('side', {
      top: topArrowStyles,
      bottom: bottomArrowStyles,
      right: rightArrowStyles,
      left: leftArrowStyles,
    })}

    border-left: ${getSpacing(1.5) as any} solid transparent;
    border-right: ${getSpacing(1.5) as any} solid transparent;
    border-bottom: calc(2 * ${getSpacing(1.5) as any} * 0.866) solid black;
    border-top: ${getSpacing(1.5) as any} solid transparent;
    display: inline-block;

    ${getCustomStyles('tooltip.styles', 'arrow')}
  }

  ${getCustomStyles('tooltip.styles', 'box')}
`;

const Tooltip = ({
  content,
  children,
  showOnClick,
  disabled = false,
  side = 'top',
  delay = 0,
  animation = SlideAnimation,
  animationProps = {},
  customStyles,
}: TooltipProps) => {
  const usedAnimationProps = { side, duration: 150, reverse: false, ...animationProps };
  const [isOpen, setOpen] = useState(false);
  const gap = parseInt((useComponentStyle('tooltip.gap') as string) || '0', 10);

  useEffect(() => {
    if (disabled || children?.props?.disabled) {
      setOpen(false);
    }
  }, [disabled, children]);

  if (!content) {
    return <>{children}</>;
  }

  return (
    <Popover
      open={isOpen && !disabled && !children?.props?.disabled}
      placement={side}
      offset={gap}
      delay={delay}
      onClick={() => showOnClick && setOpen((prevOpen) => !prevOpen)}
      onClickOutside={() => setOpen(false)}
      onMouseEnter={() => !showOnClick && setOpen(true)}
      onMouseLeave={() => !showOnClick && setOpen(false)}
      animation={animation}
      animationProps={usedAnimationProps}
      content={
        <TooltipPopover
          role="tooltip"
          data-testid="pbl-tooltip-popover"
          side={side}
          customStyles={customStyles}
        >
          <Typography variant="info">{content}</Typography>
        </TooltipPopover>
      }
    >
      {children}
    </Popover>
  );
};

export { Tooltip };
