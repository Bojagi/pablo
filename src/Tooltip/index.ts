import './styles';

export * from './Tooltip';
