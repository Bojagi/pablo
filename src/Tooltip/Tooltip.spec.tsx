import {
  render,
  getByTestId as globalGetByTestId,
  queryByTestId as globalQueryByTestId,
  getAllByTestId as globalGetAllByTestId,
  act,
  fireEvent,
  cleanup,
  waitFor,
} from '@testing-library/react';
import React from 'react';
import { PabloThemeProvider } from '../theme';
import '../../testUtils/mockResizeObserver';
import { Tooltip } from './Tooltip';

beforeEach(() => {
  jest.useFakeTimers();
  jest.spyOn(global, 'setTimeout');
});

beforeEach(() => {
  jest.spyOn(window, 'requestAnimationFrame').mockImplementation((cb) => cb(123) as any);
});

afterEach(() => {
  (window.requestAnimationFrame as any).mockRestore();
});

afterEach(cleanup);

Object.defineProperties(window.HTMLElement.prototype, {
  offsetHeight: {
    get() {
      return parseFloat(this.getAttribute('fake-height')) || 0;
    },
  },
  offsetWidth: {
    get() {
      return parseFloat(this.getAttribute('fake-width')) || 0;
    },
  },
});

test('Do not show tooltip if there is no content prop', async () => {
  const { container } = renderComponent({
    content: undefined,
  });
  // This is because of passing ref on effect, which happens on next tick
  await act(() => Promise.resolve());

  expect(container).toHaveTextContent(/^content$/);
  expect(container).toMatchSnapshot();
});

describe.each([
  ['top', 'translateY(-25%)'],
  ['right', 'translateX(12.5%)'],
  ['bottom', 'translateY(25%)'],
  ['left', 'translateX(-12.5%)'],
])('side "%s"', (side, expectedHiddenTransform) => {
  test('render and update size', async () => {
    const { container, getByTestId } = renderComponent({
      content: `This is a tooltip on the ${side} side`,
      side,
    });
    // This is because of passing ref on effect, which happens on next tick
    await act(() => Promise.resolve());

    expect(container).toMatchSnapshot();

    // Trigger resize update and update the size
    act(() => {
      getByTestId('pbl-tooltip-wrapper').setAttribute('fake-height', '10');
      getByTestId('pbl-tooltip-wrapper').setAttribute('fake-width', '20');
      fireEvent(getByTestId('pbl-tooltip-wrapper'), new Event('resize'));
    });
  });

  test('show tooltip on hover', async () => {
    const { getByTestId, queryByTestId } = renderComponent({
      content: `This is a tooltip on the ${side} side`,
      side,
    });
    // This is because of passing ref on effect, which happens on next tick
    await act(() => Promise.resolve());

    // Trigger resize update and update the size
    act(() => {
      getByTestId('pbl-tooltip-wrapper').setAttribute('fake-height', '10');
      getByTestId('pbl-tooltip-wrapper').setAttribute('fake-width', '20');
      fireEvent(getByTestId('pbl-tooltip-wrapper'), new Event('resize'));
    });

    act(() => {
      fireEvent.mouseEnter(getByTestId('pbl-tooltip-wrapper'));
    });

    await waitFor(() => {
      expect(getByTestId('pbl-animation-inner')).toHaveStyleRule('opacity', '1');
      expect(getByTestId('pbl-animation-inner')).toHaveStyleRule(
        'transform',
        'translateY(0) translateX(0)'
      );
    });

    act(() => {
      fireEvent.mouseLeave(getByTestId('pbl-tooltip-wrapper'));
    });

    await act(() => new Promise((resolve) => requestAnimationFrame(resolve as any)));

    expect(queryByTestId('pbl-tooltip-popover')).not.toBeNull();
    expect(getByTestId('pbl-animation-inner')).toHaveStyleRule('opacity', '0');
    expect(getByTestId('pbl-animation-inner')).toHaveStyleRule(
      'transform',
      expectedHiddenTransform
    );

    // At the end the popover content should be removed
    await waitFor(() => {
      expect(queryByTestId('pbl-tooltip-popover')).toBeNull();
    });
  });

  test('Do not show tooltip on hover when disabled', async () => {
    const { getByTestId, queryByTestId } = renderComponent({
      content: `This is a tooltip on the ${side} side`,
      side,
      disabled: true,
    });
    // This is because of passing ref on effect, which happens on next tick
    await act(() => new Promise((resolve) => requestAnimationFrame(resolve as any)));

    // Trigger resize update and update the size
    act(() => {
      getByTestId('pbl-tooltip-wrapper').setAttribute('fake-height', '10');
      getByTestId('pbl-tooltip-wrapper').setAttribute('fake-width', '20');
      fireEvent(getByTestId('pbl-tooltip-wrapper'), new Event('resize'));
    });

    act(() => {
      fireEvent.mouseEnter(getByTestId('pbl-tooltip-wrapper'));
      // wait for the tick to finish
      jest.advanceTimersByTime(0);
    });

    await act(() => Promise.resolve());

    expect(queryByTestId('pbl-tooltip-popover')).toBeNull();
  });

  test('Hide tooltip if button is disabled after click', async () => {
    const { getByTestId, queryByTestId, rerender } = renderComponent({
      content: `This is a tooltip on the ${side} side`,
      side,
    });
    // This is because of passing ref on effect, which happens on next tick

    await act(() => Promise.resolve());

    // Trigger resize update and update the size
    act(() => {
      getByTestId('pbl-tooltip-wrapper').setAttribute('fake-height', '10');
      getByTestId('pbl-tooltip-wrapper').setAttribute('fake-width', '20');
      fireEvent(getByTestId('pbl-tooltip-wrapper'), new Event('resize'));
    });

    act(() => {
      fireEvent.mouseEnter(getByTestId('pbl-tooltip-wrapper'));
      // wait for the tick to finish
      jest.advanceTimersByTime(500);
    });

    await waitFor(() => {
      expect(queryByTestId('pbl-tooltip-popover')).not.toBeNull();
    });

    rerender({
      content: `This is a tooltip on the ${side} side`,
      side,
      disabled: true,
    });

    act(() => {
      jest.runAllTimers();
    });

    expect(queryByTestId('pbl-tooltip-popover')).toBeNull();

    // Take the tooltip back in
    rerender({
      content: `This is a tooltip on the ${side} side`,
      side,
    });

    act(() => {
      jest.runAllTimers();
    });

    expect(queryByTestId('pbl-tooltip-popover')).toBeNull();
  });

  test('Show tooltip on click', async () => {
    const { getByTestId, queryByTestId } = renderComponent({
      content: `This is a tooltip on the ${side} side`,
      side,
      showOnClick: true,
    });

    // This is because of passing ref on effect, which happens on next tick
    await act(() => Promise.resolve());

    // Trigger resize update and update the size
    act(() => {
      getByTestId('pbl-tooltip-wrapper').setAttribute('fake-height', '10');
      getByTestId('pbl-tooltip-wrapper').setAttribute('fake-width', '20');
      fireEvent(getByTestId('pbl-tooltip-wrapper'), new Event('resize'));
    });

    act(() => {
      fireEvent.click(getByTestId('pbl-tooltip-wrapper'));
    });

    await waitFor(() => {
      expect(getByTestId('pbl-animation-inner')).toHaveStyleRule('opacity', '1');
      expect(getByTestId('pbl-animation-inner')).toHaveStyleRule(
        'transform',
        'translateY(0) translateX(0)'
      );
    });

    act(() => {
      fireEvent.click(getByTestId('pbl-tooltip-wrapper'));
    });

    await act(() => new Promise((resolve) => requestAnimationFrame(resolve as any)));

    expect(queryByTestId('pbl-tooltip-popover')).not.toBeNull();
    expect(getByTestId('pbl-animation-inner')).toHaveStyleRule('opacity', '0');
    expect(getByTestId('pbl-animation-inner')).toHaveStyleRule(
      'transform',
      expectedHiddenTransform
    );

    // At the end the popover content should be removed
    await waitFor(() => {
      expect(queryByTestId('pbl-tooltip-popover')).toBeNull();
    });
  });

  test('show tooltip on hover with delay', async () => {
    const { getByTestId, queryByTestId } = renderComponent({
      content: `This is a tooltip on the ${side} side`,
      delay: 100,
      animationDuration: 10,
      side,
    });
    // This is because of passing ref on effect, which happens on next tick
    await act(() => Promise.resolve());

    // Trigger resize update and update the size
    act(() => {
      getByTestId('pbl-tooltip-wrapper').setAttribute('fake-height', '10');
      getByTestId('pbl-tooltip-wrapper').setAttribute('fake-width', '20');
      fireEvent(getByTestId('pbl-tooltip-wrapper'), new Event('resize'));
    });

    expect(queryByTestId('pbl-animation-inner')).not.toBeInTheDocument();

    // Do mouse enter and trigger 99ms, should not make tooltip visible
    act(() => {
      fireEvent.mouseEnter(getByTestId('pbl-tooltip-wrapper'));
    });
    expect(setTimeout).toHaveBeenLastCalledWith(expect.any(Function), 100);

    expect(queryByTestId('pbl-animation-inner')).not.toBeInTheDocument();

    // Waiting 1ms less should still not make it visible
    act(() => {
      jest.advanceTimersByTime(99);
    });
    expect(queryByTestId('pbl-animation-inner')).not.toBeInTheDocument();

    // Wait remaining 1ms should make it visible
    act(() => {
      jest.advanceTimersByTime(1);
    });

    expect(getByTestId('pbl-animation-inner')).toHaveStyleRule('opacity', '1');
    expect(getByTestId('pbl-animation-inner')).toHaveStyleRule(
      'transform',
      'translateY(0) translateX(0)'
    );
  });
});

function renderComponent(props) {
  const renderFn = (innerProps) => (
    <PabloThemeProvider>
      <Tooltip animationDuration={10} {...innerProps}>
        <div data-testid="pbl-tooltip-wrapper">content</div>
      </Tooltip>
    </PabloThemeProvider>
  );

  const { baseElement, unmount, rerender } = render(renderFn(props));
  const doReRender = (innerProps) => rerender(renderFn(innerProps));
  return {
    getByTestId: globalGetByTestId.bind(null, baseElement),
    getAllByTestId: globalGetAllByTestId.bind(null, baseElement),
    queryByTestId: globalQueryByTestId.bind(null, baseElement),
    unmount,
    rerender: doReRender,
    container: baseElement,
  };
}
