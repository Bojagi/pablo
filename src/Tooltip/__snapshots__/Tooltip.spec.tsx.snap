// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Do not show tooltip if there is no content prop 1`] = `
<body>
  <div>
    <div
      data-testid="pbl-tooltip-wrapper"
    >
      content
    </div>
  </div>
</body>
`;

exports[`side "bottom" > render and update size 1`] = `
<body>
  <div>
    <div
      data-testid="pbl-tooltip-wrapper"
    >
      content
    </div>
  </div>
  <div
    data-testid="pbl-popover-mountpoint"
  />
</body>
`;

exports[`side "left" > render and update size 1`] = `
<body>
  <div>
    <div
      data-testid="pbl-tooltip-wrapper"
    >
      content
    </div>
  </div>
  <div
    data-testid="pbl-popover-mountpoint"
  />
</body>
`;

exports[`side "right" > render and update size 1`] = `
<body>
  <div>
    <div
      data-testid="pbl-tooltip-wrapper"
    >
      content
    </div>
  </div>
  <div
    data-testid="pbl-popover-mountpoint"
  />
</body>
`;

exports[`side "top" > render and update size 1`] = `
<body>
  <div>
    <div
      data-testid="pbl-tooltip-wrapper"
    >
      content
    </div>
  </div>
  <div
    data-testid="pbl-popover-mountpoint"
  />
</body>
`;
